core/worker.go

```diff
@@ -588,14 +602,9 @@ func (w *worker) commitTransaction(env *environment, tx *types.Transaction) ([]*
 		// This extra step is needed because previously the GasUsed was a public method and direct update of the value
 		// was possible.
 		env.header.SetGasUsed(gasUsed)
-
 		env.txs = append(env.txs, tx)
+		env.etxs = append(env.etxs, receipt.Etxs...)
 		env.receipts = append(env.receipts, receipt)
-		if receipt.Status == types.ReceiptStatusSuccessful {
```

Comment:
do we not need this to add it to the etxs?

Reply:
Oh yes, good catch

Reply:
fixed